GeeksCoreLibrary/Components/Repeater/Repeater.cs

```diff
@@ -318,7 +321,21 @@ private async Task<DataTable> ParseDataAsync()
                                 }
                             }
 
-                            limitClause = $" LIMIT {startIndex - totalBanners + bannersForCurrentPage}, {Settings.ItemsPerPage - bannersForCurrentPage}";
+                            if (Settings.LoadItemsUpToPageNumber)
+                            {
+                                if (loadUptoPageNumberOverrule)
```

Comment:
Kun je dit anders maken? Is nu een beetje vreemd en nu heb je dubbele code. Ten eerste is de naam verwarrend, je hebt het `loadUptoPageNumberOverrule` genoemd, maar dat is het niet. Met deze naam zou je verwachten dat je hier _altijd_ deze optie kunt overrulen, maar dat kan niet met de code die je nu hebt gemaakt; Als `Settings.LoadItemsUpToPageNumber` false is, dan doet die niets met die variabele. Graag dit een andere naam geven, of de code zo maken dat die wel altijd iets doet met die variabele.

Daarnaast heb je nu 2 keer exact dezelfde string in de limitClause, graag de code aanpassen zodat dit niet nodig is. Dit kan bijvoorbeeld ook zo bijvoorbeeld:

```c#
if (Settings.LoadItemsUpToPageNumber && !loadUptoPageNumberOverrule)
{
	limitClause = $" LIMIT 0, {Settings.ItemsPerPage * pageNumber - bannersForCurrentPage}";
}
else
{
	limitClause = $" LIMIT {startIndex - totalBanners + bannersForCurrentPage}, {Settings.ItemsPerPage - bannersForCurrentPage}";
}
````

Of om te zorgen dat die variabele echt een overrule is, kun je zoiets doen (wel even goed testen of dit werkt en niet klakkeloos overnemen of als je een betere manier weet, dat doen, ik heb dit uit mijn hoofd hier getypt):
```C#
var loadItemsUpToPageNumber = Settings.LoadItemsUpToPageNumber;
var loadUptoPageNumberOverruleValue = httpContextAccessor.HttpContext.Request.Query["loadUptoPageNumberOverrule"].ToString()
if (!String.IsNullOrWhiteSpace(loadUptoPageNumberOverruleValue))
{
	bool.TryParse(httpContextAccessor.HttpContext.Request.Query["loadUptoPageNumberOverrule"].ToString(), out loadUptoPageNumberOverruleValue);
}

if (loadUptoPageNumberOverruleValue)
{
	limitClause = $" LIMIT 0, {Settings.ItemsPerPage * pageNumber - bannersForCurrentPage}";
}
else
{
	limitClause = $" LIMIT {startIndex - totalBanners + bannersForCurrentPage}, {Settings.ItemsPerPage - bannersForCurrentPage}";
}
```